### ISSUE TITLE
Add capability to mark arguments ignored

### ISSUE THREAD

**Issue by harningt** [mpeterv/luacheck#11]
In developing some APIs, arguments to a function may not be used but are critical to matching the API the object defines.

Ex (horribly ugly to reduce line-height)
```lua
local a = { __index = {} }
local b = { __index = {} }
-- self ignored
function a.__index.apply_stuff(self, y, z)
    return y + z
end
-- z ignored
function b.__index.apply_stuff(self, y, z)
    return self.x + y
end
```

Perhaps using a _ suffix or prefix would be good to indicate parameters unused... probably via configuration option. Example w/ prefix applied:
```lua
local a = { __index = {} }
local b = { __index = {} }
-- self ignored
function a.__index.apply_stuff(_self, y, z)
    return y + z
end
-- z ignored
function b.__index.apply_stuff(self, y, _z)
    return self.x + y
end
```

**Comment by mpeterv** [comment]
I agree that this can be useful, but IMO the usecase is too small to make an option for it. In the next version(0.8.0) I plan to add a more flexible system for filtering warnings using numeric codes, then it should be possible to use something like `--ignore 212/_.*` to achieve the effect of your suggestion.(prefix version; `212` is code for unused argument warning, `_.*` is a pattern matching ignored names).

**Comment by mpeterv** [comment]
In the meantime you can use `-a`, though it filters out too many warnings (all unused arguments and loop variables).

**Comment by mpeterv** [comment]
New filtering system implemented on master branch, `--ignore 212/_.*` works. Unfortunately bleeding-edge version is not very useful now (warning codes are not documented or displayed and there are some half-implemented new warnings). I expect to release 0.8.0 anywhere from next week to the end of January.
